test/ruby/test_module.rb

```diff
@@ -1572,6 +1572,16 @@ def test_prepend
     assert_equal(expected, obj.m1)
   end
 
+  def test_prepend_block
+    C0.prepend do
```

Comment:
@simi I couldn't get it to fail in any order, but that's a great point. I modified the test to just make a dynamic class in the test rather than modifying `C0`.